### PR TITLE
PUBDEV-8483: fixed Py warnings before model training

### PR DESCRIPTION
**Pull request by sebhrusen**
https://h2oai.atlassian.net/browse/PUBDEV-8483

### CODE REVIEW

File: h2o-py/h2o/estimators/estimator_base.py [hunk 1]
```diff
@@ -57,6 +58,13 @@ class H2OEstimator(ModelBase):
     """
     
     supervised_learning = None  # overridden in implementation
+    __default_params = None     # mangled/hidden: use `_default_param_value` to access default param values internally.
+
+    def __new__(cls, *args, **kwargs):
+        if cls.__default_params is None:
+            cls.__default_params = object()  # anything but None to avoid infinite recursion
```
Comment (michalkurka) [inline h2o-py/h2o/estimators/estimator_base.py:65]: Would it be easier to create the reference instance on demand in `_check_and_save_parm`?

**Review by michalkurka — APPROVED** [review]
LGTM, thank you for addressing it quickly!

**Review by maurever — APPROVED** [review]
Looks good to me. Thanks, @sebhrusen for this fix!

File: h2o-py/h2o/estimators/estimator_base.py (hunk 1 above)
Comment (sebhrusen) [reply h2o-py/h2o/estimators/estimator_base.py:65]: yes, I succumbed to excessive refinement… will improve

**Review by koniecsveta — APPROVED** [review]

**Review by tomasfryda — APPROVED** [review]

**Pull request merged**